sklearn/linear_model/ridge.py

```diff
@@ -1217,9 +1487,14 @@ def fit(self, X, y, sample_weight=None):
                 raise ValueError("cv!=None and store_cv_values=True "
                                  " are incompatible")
             parameters = {'alpha': self.alphas}
+            solver = 'sparse_cg' if sparse.issparse(X) else 'auto'
             gs = GridSearchCV(Ridge(fit_intercept=self.fit_intercept,
-                                    normalize=self.normalize),
-                              parameters, cv=self.cv, scoring=self.scoring)
+                                    normalize=self.normalize,
+                                    solver=solver),
+                              parameters, cv=cv, scoring=self.scoring)
+            # note: unlike when using gcv, sample weights won't be used
+            # to compute the validation score so selected hyperparameter
+            # may differ
```

Comment:
Maybe you could put this note in the docstring, for instance at the end of the paragraph that documents the `sample_weights` parameter.